### PR TITLE
Add homer Integration

### PR DESCRIPTION
**Pull request by devopsec**
### What does this PR do?

Adds integration for homer HEP server.

### Motivation

No integrations for this service existed.

### Review checklist

- [x] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [x] Feature or bugfix has tests
- [x] Git history is clean
- [x] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

None.

### CODE REVIEW

**Comment by devopsec** [comment]
fixed CI build integration

**Review by buraizu — CHANGES_REQUESTED** [review]

File: homer/datadog_checks/homer/check.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        # check if homer processes are alive
+        down_services = []
+        for service in homer_service_checks:
+            if not HomerCheck.isProcAlive(service):
+                down_services.append(service)
+        if len(down_services) != 0:
+            self.service_check(
+                "services_up", AgentCheck.CRITICAL, message=f'found homer services that are down: {down_services}'
+            )
+            return None
```
Comment (FlorentClarret) [inline homer/datadog_checks/homer/check.py:58]:
```suggestion
            return
```

Also you could log the message using `self.log.error`

File: homer/datadog_checks/homer/check.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            self.service_check(
+                "metrics_up", AgentCheck.WARNING, message='agent misconfiguration: homer_data_db is not valid'
+            )
+            return None
+
+        # check if homer processes are alive
+        down_services = []
+        for service in homer_service_checks:
+            if not HomerCheck.isProcAlive(service):
+                down_services.append(service)
```
Comment (FlorentClarret) [inline homer/datadog_checks/homer/check.py:53]: Small nit, you could write this that way `down_services = [s for s in homer_service_checks if self.is_proc_alive(s)]`

File: homer/datadog_checks/homer/check.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+                        (row[0], row[1]),
+                    )
+                    stat = cur2.fetchone()[0]
+                    self.gauge(f'txcount.{row[1]}', float(stat))
+        except Exception as ex:
+            self.service_check("metrics_up", AgentCheck.WARNING, message=f'error retrieving metrics: {str(ex)}')
+            return None
+        finally:
+            conn1.close()
+            conn2.close()
```
Comment (FlorentClarret) [inline homer/datadog_checks/homer/check.py:94]: Alternatively, you could remove `ConnNoop()` and close `conn1` and `conn2` if they are not None

File: homer/datadog_checks/homer/check.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+                for row in cur1:
+                    cur2.execute(
+                        "SELECT SUM(n_live_tup) FROM pg_stat_user_tables WHERE relname LIKE CONCAT('hep_proto', '_', %s, '_', %s, '_', TO_CHAR(CURRENT_DATE, 'YYYYMMDD'), '_%%');",  # noqa: E501
+                        (row[0], row[1]),
+                    )
+                    stat = cur2.fetchone()[0]
+                    self.gauge(f'txcount.{row[1]}', float(stat))
+        except Exception as ex:
+            self.service_check("metrics_up", AgentCheck.WARNING, message=f'error retrieving metrics: {str(ex)}')
+            return None
```
Comment (FlorentClarret) [inline homer/datadog_checks/homer/check.py:91]:
```suggestion
            return
```

File: homer/datadog_checks/homer/check.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        :param name:    Process name to check
+        :type name:     str
+        :return:        Whether the process is running or not
+        :rtype:         bool
+        """
+
+        for proc in psutil.process_iter(['name']):
+            if proc.info['name'] == name:
+                return True
+        return False
```
Comment (FlorentClarret) [inline homer/datadog_checks/homer/check.py:112]: Does that mean that the agent has to run on the exact same host as your homer services?

File: homer/images/IMAGES_README.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+only hold one video.
+
+```
+File type       : MP4 H.264
+File size       : Max 1 video; 1 GB maximum size
+File dimensions : The aspect ratio must be exactly 16:9, and the resolution must be 1920x1080 or higher
+File name       : partnerName-appName.mp4
+Run time        : Recommendation of 60 seconds or less
+Description     : 300 characters maximum
+```
```
Comment (FlorentClarret) [inline homer/images/IMAGES_README.md:46]: You can remove this file

File: homer/metadata.csv [hunk 7]
```diff
@@ -0,0 +1,4 @@
+metric_name,metric_type,interval,unit_name,per_unit_name,description,orientation,integration,short_name,curated_metric
+homer.txcount.call,gauge,,,,,0,homer,homer.txcount.call,
+homer.txcount.default,gauge,,,,,0,homer,homer.txcount.default,
+homer.txcount.registration,gauge,,,,,0,homer,homer.txcount.registration,
```
Comment (FlorentClarret) [inline homer/metadata.csv:4]: Could you also provide a `description` to your metrics?

File: homer/setup.py [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+    # License
+    license='BSD-3-Clause',
+    # See https://pypi.org/classifiers
+    classifiers=[
+        'Development Status :: 5 - Production/Stable',
+        'Intended Audience :: Developers',
+        'Intended Audience :: System Administrators',
+        'Topic :: System :: Monitoring',
+        'License :: OSI Approved :: BSD License',
+        'Programming Language :: Python :: 2.7',
```
Comment (FlorentClarret) [inline homer/setup.py:66]:
```suggestion
```

File: homer/tests/test_homer.py [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+
+def test_check(dd_run_check, aggregator, instance):
+    # type: (Callable[[AgentCheck, bool], None], AggregatorStub, Dict[str, Any]) -> None
+    check = HomerCheck('homer', {}, [instance])
+    dd_run_check(check, True)
+
+    aggregator.assert_all_metrics_covered()
+    aggregator.assert_metrics_using_metadata(get_metadata_metrics())
```
Comment (FlorentClarret) [inline homer/tests/test_homer.py:22]: You can also assert the service checks

File: homer/tox.ini [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+platform = linux|darwin|win32
+extras = deps
+deps =
+    datadog-checks-base[deps]>=6.6.0
+    -rrequirements-dev.txt
+passenv =
+    DOCKER*
+    COMPOSE*
+commands =
+    pytest -v {posargs}
```
Comment (FlorentClarret) [inline homer/tox.ini:25]: You do not need a tox because you are using hatch 🙂

File: homer/manifest.json [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+        "metadata_path": "metadata.csv"
+      },
+      "service_checks": {
+        "metadata_path": "assets/service_checks.json"
+      }
+    },
+    "logs": {
+        "source": "homer"
+    }
+  },
```
Comment (FlorentClarret) [inline homer/manifest.json:41]: You should also add your dashboard in this section

**Review by FlorentClarret — CHANGES_REQUESTED** [review]
Hi! Thanks for this PR. 

I've made a first pass. Let me know if you need any help 🙂

File: homer/manifest.json [hunk 12]
```diff
@@ -0,0 +1,49 @@
+{
+  "manifest_version": "2.0.0",
+  "app_uuid": "889e9946-2b9b-451c-aacf-898180ea10cf",
+  "app_id": "homer",
+  "display_on_public_website": false,
```
Comment (FlorentClarret) [inline homer/manifest.json:5]: Is there a particular reason to set this to `false`?

**Comment by alopezz** [comment]
@devopsec Do you still intend to work on this?

**Comment by alopezz** [comment]
Closing this due to inactivity. Feel free to re-open if you start working on it again.

**Pull request closed without merging**